cmd/tempo/app/modules.go

```diff
@@ -300,12 +301,12 @@ func (t *App) initQueryFrontend() (services.Service, error) {
 
 	// http search endpoints
 	t.Server.HTTP.Handle(addHTTPAPIPrefix(&t.cfg, api.PathSearch), searchHandler)
-	t.Server.HTTP.Handle(addHTTPAPIPrefix(&t.cfg, api.PathSearchTags), searchHandler)
-	t.Server.HTTP.Handle(addHTTPAPIPrefix(&t.cfg, api.PathSearchTagsV2), searchHandler)
-	t.Server.HTTP.Handle(addHTTPAPIPrefix(&t.cfg, api.PathSearchTagValues), searchHandler)
-	t.Server.HTTP.Handle(addHTTPAPIPrefix(&t.cfg, api.PathSearchTagValuesV2), searchHandler)
+	t.Server.HTTP.Handle(addHTTPAPIPrefix(&t.cfg, api.PathSearchTags), searchTagsHandler)
+	t.Server.HTTP.Handle(addHTTPAPIPrefix(&t.cfg, api.PathSearchTagsV2), searchTagsHandler)
+	t.Server.HTTP.Handle(addHTTPAPIPrefix(&t.cfg, api.PathSearchTagValues), searchTagsHandler)
+	t.Server.HTTP.Handle(addHTTPAPIPrefix(&t.cfg, api.PathSearchTagValuesV2), searchTagsHandler)
```

Comment:
By the same logic, shouldn't we split tag value search as well?

Reply:
Before this change, recent search was going directly to querier, instead of going via search sharder (where we calculate throughout and slo metrics), and was not accounted for in slo and throughout metrics.

before this split `/search` for recent data queries (search request without start and end), and tag searchs was sent directly to a querier without going via search sharder.

for the case of tags and tag values, we want to direcly go to querier, and we do not compute throughout and slo metrics on these endpoints. And because of that I think makes sense to just have single handler for both tags and tag values endpoint.

Reply:
PR description was bit misleading, updated now.

Reply:
It took me a couple of reads, but now I understand it. Thank you for the explanation.